cylc/flow/option_parsers.py

```diff
@@ -841,21 +857,31 @@ def cleanup_sysargv(
         x.kwargs.get('dest', x.args[0].strip(DOUBLEDASH)): x
         for x in compound_script_opts
     }
-    # Filter out non-cylc-play options.
-    args = [i.split('=')[0] for i in sys.argv]
-    for unwanted_opt in (set(options.__dict__)) - set(script_opts_by_dest):
-        for arg in compound_opts_by_dest[unwanted_opt].args:
-            if arg in sys.argv:
-                index = sys.argv.index(arg)
+
+    # Filter out non-cylc-play options:
+    # The set of options which we want to weed out:
+    for unwanted_dest in (set(options.__dict__)) - set(script_opts_by_dest):
+
+        # The possible ways this could be written - if the above
+        # were "workflow_name" this could be '-n' or '--workflow-name':
+        for unwanted_arg in compound_opts_by_dest[unwanted_dest].args:
+
+            # Check for args which are standalone or space separated
+            # `--workflow-name foo`:
+            if unwanted_arg in sys.argv:
+                index = sys.argv.index(unwanted_arg)
                 sys.argv.pop(index)
                 if (
-                    compound_opts_by_dest[unwanted_opt].kwargs['action']
+                    compound_opts_by_dest[unwanted_dest].kwargs['action']
                     not in ['store_true', 'store_false']
                 ):
                     sys.argv.pop(index)
-            elif arg in args:
-                index = args.index(arg)
-                sys.argv.pop(index)
+
+            # Check for `--workflow-name=foo`:
+            elif unwanted_arg in [a.split('=')[0] for a in sys.argv]:
```

Comment:
Recomputing this list for each "unwanted" option is a bit expensive.

It might be easier to loop over `sys.argv` than the "unwanted_opts" because this allows you to work over the opts/args in order, it also saves popping items off of the list.

I think something like this should work (very untested!):

```python
# options which require a value e.g. "--foo=a" or "--foo a"
unwanted_compound_opts: List[str]
# options which don't require a value e.g. "--foo"
unwanted_simple_opts: List[str]

pop_next = False
args = []
for arg in sys.argv:
    if pop_next:
        pop_next = False
        continue
    parts = arg.split('=', 1)
    elif parts[0] in unwanted_compound_opts:
       # we don't want this option or its value
        if len(parts) == 1:
            # no "=" in arg, skip the next arg
            pop_next = True
    elif parts[0] in unwanted_simple_opts:
       # we don't want this option
        continue
    else:
        # keep this arg
        args.append(arg)

sys.argv = args 
```